### PR TITLE
Crt http integration #2293  rebased

### PR DESCRIPTION
**Pull request by SergeyRyabinin**
*Issue #, if available:*
A rebase of
Crt http integration #2293 
*Description of changes:*

*Check all that applies:*
- [x] Did a review by yourself.
- [ ] Added proper tests to cover this PR. (If tests are not applicable, explain.)
- [x] Checked if this PR is a breaking (APIs have been changed) change.
- [x] Checked if this PR will _not_ introduce cross-platform inconsistent behavior.
- [x] Checked if this PR would require a ReadMe/Wiki update.

Check which platforms you have built SDK on to verify the correctness of this PR.
- [x] Linux
- [x] Windows
- [ ] Android
- [x] MacOS
- [ ] IOS
- [ ] Other Platforms

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by sbiscigl — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -33,6 +33,7 @@ if (LEGACY_BUILD)
     option(BUILD_SHARED_LIBS "If enabled, all aws sdk libraries will be build as shared objects; otherwise all Aws libraries will be built as static objects" ON)
     option(FORCE_SHARED_CRT "If enabled, will unconditionally link the standard libraries in dynamically, otherwise the standard library will be linked in based on the BUILD_SHARED_LIBS setting" ON)
     option(SIMPLE_INSTALL "If enabled, removes all the additional indirection (platform/cpu/config) in the bin and lib directories on the install step" ON)
+    option(USE_CRT_HTTP_CLIENT "If enabled, The common runtime HTTP client will be used, and the legacy systems such as WinHttp and libcurl will not be built or included" OFF)
```
Comment (sdavtaker) [inline CMakeLists.txt:36]: Typo after comma.
> If enabled, The common runtime HTTP 
should be 
> If enabled, the common runtime HTTP

File: src/aws-cpp-sdk-core/source/Aws.cpp [hunk 2]
```diff
@@ -16,6 +16,8 @@
 #include <aws/core/config/AWSProfileConfigLoader.h>
 #include <aws/core/internal/AWSHttpResourceClient.h>
 
+#include <aws/crt/Api.h>
```
Comment (sdavtaker) [inline src/aws-cpp-sdk-core/source/Aws.cpp:19]: Should we wrap this in conditional compile path?

File: tests/aws-cpp-sdk-transcribestreaming-integration-tests/TranscribeTests.cpp [hunk 3]
```diff
@@ -46,7 +46,7 @@ class TranscribeStreamingTests : public ::testing::Test
 #endif
         m_client = Aws::MakeUnique<TranscribeStreamingServiceClient>(ALLOC_TAG, config);
         m_clientWithWrongCreds = Aws::MakeUnique<TranscribeStreamingServiceClient>(ALLOC_TAG, Aws::Auth::AWSCredentials("a", "b"), config);
-        config.endpointOverride = "https://0xxxabcdefg123456789.com";
+        //config.endpointOverride = "https://0xxxabcdefg123456789.com";
```
Comment (sdavtaker) [inline tests/aws-cpp-sdk-transcribestreaming-integration-tests/TranscribeTests.cpp:49]: if no longer used, we should remove this line

File: src/aws-cpp-sdk-core/source/Aws.cpp (hunk 2 above)
Comment (SergeyRyabinin) [reply src/aws-cpp-sdk-core/source/Aws.cpp:19]: that's "AWS-CRT-CPP" wrapper, not the http client from CRT.

**Review by sdavtaker — APPROVED** [review]

**Pull request merged**